### PR TITLE
Optimize and harden `EncoreHelper` a bit

### PR DESCRIPTION
**Pull request by jdreesen**
## Changes in this pull request  
- throw when file does not exist or contains invalid json
- just iterate the array (instead of its keys)
- perform expensive `array_merge()` after the loop

### <samp>🤖 Generated by Copilot at 7bc496d</samp>

Improved error handling and performance of `getBuildPathsFromEntrypoints` function in `EncoreHelper.php`. The function now checks for the existence and validity of the entrypoints file and avoids unnecessary array merges.

### <samp>🤖 Generated by Copilot at 7bc496d</samp>

> _`getBuildPathsFrom`_
> _Entrypoints file, or error_
> _Fall leaves, no assets_

### <samp>🤖 Generated by Copilot at 7bc496d</samp>

* Fix bug that caused exception when entrypoints file was missing or invalid by checking if the file exists and is readable before parsing it ([link](https://github.com/pimcore/pimcore/pull/15239/files?diff=unified&w=0#diff-ae46b2c34891ec99e8400db580aa227f8391fe9ac7cea54abacc35bb79b1fb21L23-R35))
* Optimize array merging logic by using `array_replace_recursive` instead of nested loops and `array_merge` ([link](https://github.com/pimcore/pimcore/pull/15239/files?diff=unified&w=0#diff-ae46b2c34891ec99e8400db580aa227f8391fe9ac7cea54abacc35bb79b1fb21L23-R35))

### CODE REVIEW

**Comment by jdreesen** [comment]
Funny Copilot description:
> Optimize array merging logic by using array_replace_recursive instead of nested loops and array_merge ([link](https://github.com/pimcore/pimcore/pull/15239/files?diff=unified&w=0#diff-ae46b2c34891ec99e8400db580aa227f8391fe9ac7cea54abacc35bb79b1fb21L23-R35))

because it... doesn't use `array_replace_recursive()` now 😆

File: lib/Helper/EncoreHelper.php [hunk 1]
```diff
@@ -20,15 +20,18 @@ final class EncoreHelper
...
+
         $entrypointsContent = file_get_contents($entrypointsFile);
-        $entrypointsJson = json_decode($entrypointsContent, true)['entrypoints'];
-        $entrypoints = array_keys($entrypointsJson);
+        $entrypoints = json_decode($entrypointsContent, true, flags: JSON_THROW_ON_ERROR)['entrypoints'];
 
         $paths = [];
         foreach ($entrypoints as $entrypoint) {
-            $paths = array_merge($paths, $entrypointsJson[$entrypoint][$type] ?? []);
+            $paths = $entrypoint[$type] ?? [];
```
Comment (mattamon) [inline lib/Helper/EncoreHelper.php]:
```suggestion
            $paths[] = $entrypoint[$type] ?? [];
```
Maybe I don't get it but there should be brackets right? :D

Comment (jdreesen) [reply lib/Helper/EncoreHelper.php]: Oh, yes, sure, thx!

**Comment by jdreesen** [comment]
I rebased again. Pipeline is green now.

**Review by fashxp — APPROVED** [review]

**Pull request merged**